tests/VCR/Storage/AbstractStorageTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace VCR\Storage;
+
+use org\bovigo\vfs\vfsStream;
+
+/**
+ * Test integration of PHPVCR with PHPUnit.
+ */
+class AbstractStorageTest extends \PHPUnit_Framework_TestCase
+{
+    protected $handle;
+    protected $filePath;
+    protected $storage;
+
+    public function testRootNotExisting()
+    {
+        $fs = vfsStream::setup('test');
+
+        $this->storage = new TestStorage(vfsStream::url('test/'), 'file');
+        $this->assertTrue($fs->hasChild('file'));
+
+        $this->storage = new TestStorage(vfsStream::url('test/'), 'folder/file');
+        $this->assertTrue($fs->hasChild('folder'));
+        $this->assertTrue($fs->getChild('folder')->hasChild('file'));
+    }
+
+    /**
+     * Expect error
+     */
+    public function testFilePathCreated()
+    {
+        vfsStream::setup('test');
+        $this->storage = new TestStorage(vfsStream::url('test/foo'), 'file');
+    }
+
+}
+
+class TestStorage extends AbstractStorage
```

Comment:
i hope its ok for you to have this class right in the test case file. imho it makes sense to keep them here, its "local" to just this test.